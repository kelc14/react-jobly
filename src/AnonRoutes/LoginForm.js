import React, { useState } from "react";

import "./LoginForm.css";

const LoginForm = () => {
  const INITIAL_STATE = { username: "", password: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);

  /** Send {USERNAME, PASSWORD} to API to check if logged in and provide feedback
   *    & clear form. */

  const handleSubmit = (evt) => {
    evt.preventDefault();
    console.log(formData);
    setFormData(INITIAL_STATE);
  };

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** Render:
   *
   */
  return (
    <div className="LoginForm" onSubmit={handleSubmit}>
      <h2 className="LoginForm-heading">Log In</h2>
      <form className="LoginForm-form">
        <label htmlFor="username" className="LoginForm-label">
          Username:
        </label>
        <input
          id="username"
          type="text"
          name="username"
          onChange={handleChange}
          value={formData.username}
          className="LoginForm-input"
        />

        <label htmlFor="password" className="LoginForm-label">
          Password:
        </label>
        <input
          id="password"
          type="password"
          name="password"
          onChange={handleChange}
          value={formData.password}
          className="LoginForm-input"
        />

        <button className="LoginForm-button">Log in</button>
      </form>
    </div>
  );
};

export default LoginForm;
