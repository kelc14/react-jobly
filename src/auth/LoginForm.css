.LoginForm {
  width: 50vw;
  background-color: white;
  border: 1px solid black;
  border-radius: 10px;
  margin: 0 auto;
  margin-top: 10%;
  padding: 20px;
  overflow: hidden;
}
.LoginForm-heading {
  color: goldenrod;
}
.LoginForm-form {
  text-align: left;
}
.LoginForm-label {
  display: block;
  padding-top: 10px;
  font-size: 1rem;
  font-weight: 500;
}
.LoginForm-input {
  width: 100%;
  padding: 12px 20px;
  margin: 8px 0;
  box-sizing: border-box;
}
.LoginForm-button {
  background-color: goldenrod;
  border: none;
  border-radius: 10px;
  color: white;
  padding: 16px 32px;
  font-size: 1rem;
  margin: 4px 2px;
  cursor: pointer;
}
