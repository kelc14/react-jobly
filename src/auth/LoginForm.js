import React, { useState } from "react";
import JoblyApi from "../api/api";
import "./LoginForm.css";
import useErrors from "../hooks/useErrors";
import { useNavigate } from "react-router-dom";

const LoginForm = ({ loginUser }) => {
  const INITIAL_STATE = { username: "", password: "" };
  const [formData, setFormData] = useState(INITIAL_STATE);
  let navigate = useNavigate();

  const [err, documentErrors, showFormError] = useErrors();

  /** Send {USERNAME, PASSWORD} to API to check if logged in and provide feedback
   *    & clear form. */

  const handleSubmit = async (evt) => {
    evt.preventDefault();
    try {
      let token = await JoblyApi.userLogin(formData);
      console.log("token", token);
      if (token) loginUser(formData.username, token);
      navigate("/");
      setFormData(INITIAL_STATE);
    } catch (err) {
      documentErrors(err);
    }
  };

  /** setLoggedInContext */

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** Render:
   *
   */
  return (
    <div className="LoginForm" onSubmit={handleSubmit}>
      <h2 className="LoginForm-heading">Log In</h2>
      <form className="LoginForm-form">
        <label htmlFor="username" className="LoginForm-label">
          Username:
        </label>
        <input
          id="username"
          type="text"
          name="username"
          onChange={handleChange}
          value={formData.username}
          className="LoginForm-input"
          autoComplete="username"
        />

        <label htmlFor="password" className="LoginForm-label">
          Password:
        </label>
        <input
          id="password"
          type="password"
          name="password"
          onChange={handleChange}
          value={formData.password}
          className="LoginForm-input"
          autoComplete="current-password"
        />
        {err && showFormError()}

        <button className="LoginForm-button">Log in</button>
      </form>
    </div>
  );
};

export default LoginForm;
