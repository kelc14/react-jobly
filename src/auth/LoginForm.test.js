import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { MemoryRouter, Router } from "react-router-dom";
import LoginForm from "./LoginForm";
import { createMemoryHistory } from "history";

/**  SMOKE TEST */
it("renders without crashing", function () {
  render(
    <MemoryRouter>
      {" "}
      <LoginForm />
    </MemoryRouter>
  );
});

/**  SNAPSHOT TEST */
it("renders without crashing", function () {
  const { asFragment } = render(
    <MemoryRouter>
      {" "}
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

/** LOGIN FORM e2e */

it("logs a user in", function async() {
  const loginUser = jest.fn();
  render(
    <MemoryRouter>
      <LoginForm loginUser={loginUser} />
    </MemoryRouter>
  );

  const usernameInput = screen.getByLabelText("Username:");
  const passwordInput = screen.getByLabelText("Password:");
  const loginBtn = screen.getByText("Log in");

  userEvent.type(usernameInput, "testuser");
  userEvent.type(passwordInput, "password");
  userEvent.click(loginBtn);

  //   expect(loginUser).toHaveBeenCalledWith({
  //     username: "testuser",
  //     password: "password",
  //   });

  /// this is where I am stuck
});
