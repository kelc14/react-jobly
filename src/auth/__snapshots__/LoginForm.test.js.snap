// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
   
  <div
    class="LoginForm"
  >
    <h2
      class="LoginForm-heading"
    >
      Log In
    </h2>
    <form
      class="LoginForm-form"
    >
      <label
        class="LoginForm-label"
        for="username"
      >
        Username:
      </label>
      <input
        autocomplete="username"
        class="LoginForm-input"
        id="username"
        name="username"
        type="text"
        value=""
      />
      <label
        class="LoginForm-label"
        for="password"
      >
        Password:
      </label>
      <input
        autocomplete="current-password"
        class="LoginForm-input"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        class="LoginForm-button"
      >
        Log in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
